File: app_project/backend/app/schemas/news/news.py
from pydantic import BaseModel
from datetime import datetime

class NewsCategoryBase(BaseModel):
    name: str
    slug: str
    description: str | None = None

class NewsCategoryCreate(NewsCategoryBase):
    pass

class NewsCategoryUpdate(NewsCategoryBase):
    pass

class NewsCategoryResponse(NewsCategoryBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class NewsArticleBase(BaseModel):
    title: str
    slug: str
    summary: str | None = None
    content: str
    is_published: bool = False

class NewsArticleCreate(NewsArticleBase):
    category_id: int
    image: bytes | None = None

class NewsArticleUpdate(NewsArticleBase):
    category_id: int | None = None
    image: bytes | None = None

class NewsArticleResponse(NewsArticleBase):
    id: int
    category_id: int
    published_at: datetime
    created_at: datetime
    updated_at: datetime
    image: bytes | None = None

    class Config:
        from_attributes = True

File: app_project/frontend/src/models/news/news.ts
export interface NewsCategoryBase {
  name: string;
  slug: string;
  description?: string | null;
}

export interface NewsCategoryCreate extends NewsCategoryBase {}
export interface NewsCategoryUpdate extends NewsCategoryBase {}

export interface NewsCategoryResponse extends NewsCategoryBase {
  id: number;
  created_at: string;
  updated_at: string;
}

export interface NewsArticleBase {
  title: string;
  slug: string;
  summary?: string | null;
  content: string;
  is_published?: boolean;
}

export interface NewsArticleCreate extends NewsArticleBase {
  category_id: number;
  image?: File | null;
}

export interface NewsArticleUpdate extends NewsArticleBase {
  category_id?: number;
  image?: File | null;
}

export interface NewsArticleResponse extends NewsArticleBase {
  id: number;
  category_id: number;
  published_at: string;
  created_at: string;
  updated_at: string;
  image?: string | null; 
}
